MapboxNavigation/UILabel.swift

```diff
@@ -0,0 +1,13 @@
+import Foundation
+
+protocol Abridgeable: class {
+    var unabridgedText: String? { get set }
+    typealias AvailableBoundsHandler = () -> (CGRect)
+    var availableBounds: AvailableBoundsHandler! { get set }
+}
+
+extension Abridgeable where Self: UILabel {
+    func set(unabridgedText: String?) {
```

Comment:
Is StreetLabel the only class we foresee conforming to Abridgeable? If so, maybe we could sidestep these protocol complications by folding `unabridgedText` into StreetLabel. Then we could implement `didSet` on `bounds` to make sure `text` gets updated without ever having to trigger an update explicitly.

Otherwise, if other UILabel subclasses will need to be abridgeable, we could fold `unabridgedText` into StyleLabel, then use a `abridgesText` flag to determine whether the `bounds` `didSet` calls `abbreviated(toFit:)` or sets `text` to `unabridgedText` directly.